Ticket 33617 (#1808)
torproject/tor: doc/tor.1.txt

extra-info document. The following options affect the different
types of statistics that Tor relays collect and publish:

> [[BandwidthStatistics]] **BandwidthStatistics** **0**|**1**::

Author: Please document that this option takes "auto".

    Relays and bridges only.
    When this option is enabled, Tor collects statistics about bandwidth
    used each 24 hours for the last 5 days (if available). Reports
    bandwidth statistics for OR and directory reads and writes,
    and writes them to disk every 24 hours. Enables relay and bridge operators
    to monitor bandwidth used by Tor. If ExtraInfoStatistics is enabled,

>     it will be published as part of the extra-info document. (Default: 1)

Author: Please describe what "auto" does. Check the rest of the manual page for examples.

extra-info document. The following options affect the different
types of statistics that Tor relays collect and publish:

[[BandwidthStatistics]] **BandwidthStatistics** **0**|**1**::

>     Relays and bridges only

Author: Consistency
```suggestion
    Relays and bridges only.
```

    When this option is enabled, Tor collects statistics about bandwidth

>     used each 24 hours for the last 120 hours (if available). Reports

Author: Readability
```suggestion
    used each 24 hours for the last 5 days (if available). Reports
```

    bandwidth statistics for read and write operations and read and write
    operations for directory protocol and writes them in disk every 24

>     24 hours. Enables relay and bridge operators to monitor bandwidth

Author: Readability - shorter sentences
```suggestion
    bandwidth statistics for read and write operations (including the OR and directory protocols)
    and writes them to disk every 24 hours. Enables relay and bridge operators to monitor bandwidth
```

    used by Tor. If ExtraInfoStatistics is enabled, it will be published

>     as part of extra-info document. (Default: 1)

Author: Typo
```suggestion
    as part of the extra-info document. (Default: 1)
```

>     published as part of the extra-info document. (Default: auto)

Author: This man page entry doesn't answer these questions:
* What happens to statistics collection if the BandwidthStatistics option is auto ?
* Are statistics published when the BandwidthStatistics option is 1 ?
* Are statistics published when the BandwidthStatistics option is 0 ?
* Are statistics published when the BandwidthStatistics consensus parameter is auto?

Please rewrite the consensus parameter part of the man page entry. Use another consensus parameter man page entry as a template:
https://github.com/torproject/tor/blob/master/doc/tor.1.txt#L2766

extra-info document. The following options affect the different
types of statistics that Tor relays collect and publish:

> [[BandwidthStatistics]] **BandwidthStatistics** **0**|**1**|**auto**::
>     Relays and bridges only. 
> 	When this option is enabled or left unspecified, Tor collects
>     statistics about bandwidth used each 24 hours for the
>     last 5 days (if available) and writes them to disk every 24 hours.
>     Enables relay and bridge operators to monitor bandwidth used by Tor.
>     Tor includes the bandwidth statistcs in extrainfo document if and
>     only if BandwidthStatistics is enabled or left unspecified,
>     in which case the consensus parameter, BandwidthStatistics,
>     is checked further whether it is enabled or not.
>     If ExtraInfoStatistics is enabled, it will published as part of
>     extra-info document. (Default: auto)

Author: We've tried revising this man page entry a few times already. So I'm going to show you how to create a man page entry, using another man page entry as a template.

Take a copy of a similar man page entry. Let's use CellStatistics:
```
[[CellStatistics]] **CellStatistics** **0**|**1**::
    Relays only.
    When this option is enabled, Tor collects statistics about cell
    processing (i.e. mean time a cell is spending in a queue, mean
    number of cells in a queue and mean number of processed cells per
    circuit) and writes them into disk every 24 hours. Onion router
    operators may use the statistics for performance monitoring.
    If ExtraInfoStatistics is enabled, it will published as part of
    extra-info document. (Default: 0)
```

Change the copy so that it documents the new option. Only make the changes that are needed:
```
[[BandwidthStatistics]] **BandwidthStatistics** **0**|**1**|**auto**::
    Relays and bridges only.
    When this option is enabled, Tor collects statistics about bandwidth
    (i.e. read and written bytes, and read and written directory bytes) and
    writes them into disk every 24 hours. Relay and bridge
    operators may use the statistics for performance monitoring.
    If ExtraInfoStatistics is enabled, it will published as part of
    the extra-info document. (Default: auto)
```

This option also has a consensus parameter. So we also need to find a man page entry that documents a consensus parameter. Let's use the consensus parameter sentence from NumDirectoryGuards:
```
    If this option is set to 0, use the
    value from the guard-n-primary-dir-guards-to-use consensus parameter, and
    default to 3 if the consensus parameter isn't set. 
```

Change the copy so that it documents the new consensus parameter. Only make the changes that are needed:
```
    If this option is set to "auto", use the value from the BandwidthStatistics
    consensus parameter, and default to 1 if the consensus parameter isn't set.
```

Now, combine the two parts into a man page entry, and wrap the lines:
```suggestion
[[BandwidthStatistics]] **BandwidthStatistics** **0**|**1**|**auto**::
    Relays and bridges only.
    When this option is enabled, Tor collects statistics about bandwidth (i.e.
    read and written bytes, and read and written directory bytes) and writes
    them to disk every 24 hours. Relay and bridge operators may use the
    statistics for performance monitoring. If this option is set to "auto",
    use the value from the BandwidthStatistics consensus parameter, and
    default to 1 if the consensus parameter isn't set. If ExtraInfoStatistics
    is enabled, it will published as part of the extra-info document.
    (Default: auto)
```

Sometimes, fewer changes are better.

    Only used by servers. This file is used to collect approximate connection
    history (number of active connections over time).

__DataDirectory__/**`stats/bandwidth-stats`**::

>     Only used by servers.This file is used collect information about

Author: Typo
```suggestion
    Only used by servers. This file is used collect information about
```

Author: Looks like you missed this typo?
